index.js

```diff
@@ -8,62 +8,36 @@ var platforms = {
 };
 
 
-exports.verifyPayment = function (platform, payment, cb) {
-	function syncError(error) {
-		process.nextTick(function () {
-			cb(error);
-		});
-	}
-
-	if (!payment) {
-		return syncError(new Error('No payment given'));
-	}
-
-	var engine = platforms[platform];
-
-	if (!engine) {
-		return syncError(new Error('Platform ' + platform + ' not recognized'));
-	}
-
-	engine.verifyPayment(payment, function (error, result) {
-		if (error) {
-			return cb(error);
+function makeMethod(methodName) {
+	exports[methodName] = function (platform, payment, cb) {
+		function asyncError(message) {
```

Comment:
Do we still need this function at all if we could just `process.nextTick(cb, error)` everywhere?

Reply:
it slightly reduces boilerplate at the places where errors are created, compare:

```js
asyncError('Platform ' + platform + ' not recognized')
```

with

```js
process.nextTick(cb, new Error('Platform ' + platform + ' not recognized'))
```

but I can remove if you prefer

Reply:
It would also void the need of `captureStackTrace`, right? So it's kinda 2 birds 1 stone, I think. I would prefer the latter (straight to nextTick), unless anyone has any strong reasons why we can't.

Reply:
up to you man, its not my project ;)